db/migrate/20150204143153_correct_branch_code_index.rb

```diff
@@ -0,0 +1,11 @@
+class CorrectBranchCodeIndex < ActiveRecord::Migration
+  def change
+    # This index was originally unique
+    # Which resulted in it only being possible to have ONE branch 'B1S'
+    # in the entire table
+    #
+    # see 20140322201843_create_branches.rb
+    remove_index :branches, :code
+    add_index :branches, :code
```

Comment:
Wow that was stupid, the index was supposed to be with the tree. Do you reckon we could do it properly (I don't mind if you prefer to do validation outside of the database, as long as we have plenty of tests). E.g.

    add_index :branches, [:tree_id, :code], unique: true

:+1: